Makefile

```diff
@@ -20,18 +20,12 @@ include ./mk/pretty.mk
 
 # Run in interactive and parallel mode by default
 
-# You can use the $(PARALLEL_TESTS_FILE) file for setting the number of
-# parallel tests, e.g.
-#   PARALLEL_TESTS = 123
-
-PARALLEL_TESTS_FILE = mk/parallel-tests.mk
-
-ifeq ($(wildcard $(PARALLEL_TESTS_FILE)),)
-# Setting the default value.
-PARALLEL_TESTS = $(shell getconf _NPROCESSORS_ONLN)
-else
-# Getting the value from the $(PARALLEL_TESTS_FILE) file.
-include $(PARALLEL_TESTS_FILE)
+# You can use the PARALLEL_TESTS variable to control the number of parallel
```

Comment:
I wonder whether this file `mk/parallel-tests.mk` wasn't a convenience to define the number of parallel test on your system according to your liking once and for all.

How would you do this now?

Reply:
You can add it to `mk/config.mk` along with any other custom overriden variables. That file is loaded if it exists and seemed a better solution than a file per variable.

Or to make this behave exactly as before:
```sh
echo '-include $(TOP)/mk/parallel-tests.mk' >> mk/config.mk
```

`mk/parallel-tests.mk` was undocumented, and scripts did not reference it, so I believe if it's used it's most likely by the people reviewing this 🙂 But I'd be  happy to send a message to the list if it's more widely known.

Reply:
Local configuration should be in files that are not under version control, as git cannot handle well situations where you make changes to registered files that you never want to push.
If I change `config.mk`, git will annoy me to death.  (`darcs` could deal with commits one never wants to push.)

Reply:
> But I'd be happy to send a message to the list if it's more widely known.

I know and use it.

Reply:
@andreasabel — The path `mk/config.mk` is already `.gitignored`. It does not exist under version control in the same way that `mk/parallel-tests.mk` did not. (`mk/config.mk` was previously never loaded, but now is loaded if it exists).

But I'll restore the loading of `mk/parallel-test.mk` as well since it sounds like it's a concern. It just seemed better to nip the single instance of the "one file per variable" pattern while there was only one case of it.

Reply:
If `mk/config.mk` is already `.gitignored` I have no problem on using it instead of `mk/parallel-tests.mk`.

Reply:
I'm doing a couple things to help with this:
  1. Changing the name `mk/config.mk` to `mk/custom.mk` to reinforce that this path is `.gitignored`. (Again, the path was not loaded before this PR).
  1. If `mk/config.mk` does not exist and `mk/parallel-tests.mk` does, emit a small warning + hint and load it. This will prevent any surprising changes for developers. Hopefully that block can be removed in the future.
  2. I'm adding both the existence + purpose of `mk/config.mk` and of the `PARALLEL_TESTS` variable to the `HACKING.md`, so that it's more discoverable.

Reply:
I misclick-submitted that comment too early. Although `mk/custom.mk` is a better name, the benefit of `mk/config.mk` is that it was _already_ ignored, so would not show up in branch changes. So scratch that first point.

Reply:
That's now pushed

Reply:
Good, my concern has been addressed!